qmp/socket.go

```diff
@@ -48,9 +47,12 @@ type SocketMonitor struct {
 	// Send command responses and errors
 	stream <-chan streamResponse
 
-	// Send domain events to listeners when available
-	listeners *int32
-	events    <-chan Event
+	// Protects accesses against the noListeners chan since it is
+	// swapped out in the call to SocketMonitor.Events.
+	lmu sync.Mutex
```

Comment:
I kept waffling on whether this mutex was necessary or helpful and ended up just leaving it in to err on the side of caution.